### PR TITLE
allow limiting access by passwords

### LINKED ISSUE #194: Restrict downloads with passwords
While you don't want to include a one time secret (#184) just yet, it'd be nice if there was a possibility to restrict downloads with passwords. Currently I'm using several linx installations for several clients of mine with a .htaccess file for the nginx reverse proxy in front of them, it'd be awesome if that could be done in linx itself.

### PR DESCRIPTION
**Pull request by stek29**
fixes #194

### CODE REVIEW

**Comment by stek29** [comment]
@andreimarcu if you want some high-level overview of my changes tell me

**Comment by stek29** [comment]
Hello?..

**Comment by andreimarcu** [comment]
I'll give it a look hopefully in the next few days. Thanks for your patience

File: access.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		return
+	} else if err != nil {
+		oopsHandler(c, w, r, RespAUTO, "Corrupt metadata.")
+		return
+	}
+
+	if src, err := checkAccessKey(r, &metadata); err != nil {
+		// remove invalid cookie
+		if src == accessKeySourceCookie {
+			setAccessKeyCookies(w, getSiteURL(r), fileName, "", time.Unix(0, 0))
```
Comment (andreimarcu) [inline access.go]: You're passing getSiteURL where you need the domain instead ("https://domain.tld/" vs "domain.tld") which causes the following: 
```
net/http: invalid Cookie.Domain "http://localhost:8080/"; dropping domain attribute
```

File: access.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+		return
+	}
+
+	if metadata.AccessKey != "" {
+		var expiry time.Time
+		if Config.accessKeyCookieExpiry != 0 {
+			expiry = time.Now().Add(time.Duration(Config.accessKeyCookieExpiry) * time.Second)
+		}
+		setAccessKeyCookies(w, getSiteURL(r), fileName, metadata.AccessKey, expiry)
```
Comment (andreimarcu) [inline access.go]: same as above

**Comment by stek29** [comment]
I also haven't updated API docs.

**Comment by stek29** [comment]
@andreimarcu 

1. cookie doman and path were fixed
1. password input box on upload page is now hidden with checkbox
1. password input box is added to paste too
1. page for accessing a password-protected file is wrapped in `div#main`, so it has white background now
1. access key header is now mentioned on API docs page

**Comment by andreimarcu** [comment]
Thank you for your work!

**Pull request merged**